### PR TITLE
picocrypt: init at 1.34

### LINKED ISSUE #321719: Package request: picocrypt
**Project description**

 A very small, very simple, yet very secure encryption tool. 

**Metadata**

* homepage URL: https://github.com/Picocrypt/Picocrypt
* source URL: https://github.com/Picocrypt/Picocrypt
* license: gpl3
* platforms: unix, linux, darwin, windows

### PR DESCRIPTION
**Pull request by ryand56**
## Description of changes

Adds [Picocrypt](https://github.com/Picocrypt/Picocrypt), a lightweight file encryption tool. Let me know if there is anything I need to change as this is my first PR.
Closes #321719

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/pi/picocrypt/package.nix [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    xorg.libXi.dev
+    xorg.libXinerama.dev
+    xorg.libXxf86vm
+    libGL.dev
+    gtk3
+  ];
+
+  nativeBuildInputs = [
+    copyDesktopItems
+    pkg-config
```
Comment (Aleksanaa) [inline pkgs/by-name/pi/picocrypt/package.nix:46]:
```suggestion
    pkg-config
    wrapGAppsHook3
```

You never know when will others request you to add it. Just add it in case of gtk3/4.

File: pkgs/by-name/pi/picocrypt/package.nix [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  src = fetchFromGitHub {
+    owner = "Picocrypt";
+    repo = "Picocrypt";
+    rev = version;
+    hash = "sha256-TO72s8v0cpyKjvi0b74vux3+VzTfW540Drtr2bD5xVw=";
+  };
+
+  sourceRoot = "${src.name}/src";
+
+  vendorHash = "sha256-W982HiosXvDadMJJ0wP6AsalQ/uxklSbbmFp26XQEhM=";
```
Comment (Aleksanaa) [inline pkgs/by-name/pi/picocrypt/package.nix:27]:
```suggestion
  vendorHash = "sha256-W982HiosXvDadMJJ0wP6AsalQ/uxklSbbmFp26XQEhM=";

```

File: maintainers/maintainer-list.nix
Comment (Aleksanaa) [inline maintainers/maintainer-list.nix:1]: Please split this into another commit: `maintainers: add ryand56`.

File: pkgs/by-name/pi/picocrypt/package.nix [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    wrapGAppsHook3
+  ];
+
+  CGO_ENABLED = 1;
+
+  postInstall = ''
+    mv $out/bin/Picocrypt $out/bin/picocrypt-gui
+    d=$out/share/icons/hicolor/scalable/apps/
+    mkdir -p $d
+    cp $src/images/key.svg $d/picocrypt.svg
```
Comment (Aleksanaa) [inline pkgs/by-name/pi/picocrypt/package.nix:58]:
```suggestion
    install -Dm644 -t $out/share/icons/hicolor/scalable/apps images/key.svg 
```

The above two lines can be removed.

**Review by 0x61nas — APPROVED** [review]

File: pkgs/by-name/pi/picocrypt/package.nix [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    copyDesktopItems
+    pkg-config
+    wrapGAppsHook3
+  ];
+
+  CGO_ENABLED = 1;
+
+  postInstall = ''
+    mv $out/bin/Picocrypt $out/bin/picocrypt-gui
+    install -Dm644 -t $out/share/icons/hicolor/scalable/apps images/key.svg
```
Comment (Aleksanaa) [inline pkgs/by-name/pi/picocrypt/package.nix:56]:
```suggestion
    install -Dm644 images/key.svg $out/share/icons/hicolor/scalable/apps/picocrypt.svg
```

Sorry, it was wrong

Comment (ryand56) [reply pkgs/by-name/pi/picocrypt/package.nix]: Okay, fixed

File: pkgs/by-name/pi/picocrypt/package.nix [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  ];
+
+  meta = {
+    description = "Very small, very simple, yet very secure encryption tool, written in Go";
+    homepage = "https://github.com/Picocrypt/Picocrypt";
+    changelog = "https://github.com/Picocrypt/Picocrypt/blob/main/Changelog.md";
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [ ryand56 ];
+    mainProgram = "picocrypt-gui";
+    platforms = [ "x86_64-linux" ];
```
Comment (Aleksanaa) [inline pkgs/by-name/pi/picocrypt/package.nix:77]:
```suggestion
```

(Why are there always things I don't see)

Just let it build. It cannot harm you.

**Comment by nadir-ishiguro** [comment]
Result of `nixpkgs-review pr 322846` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>picocrypt</li>
  </ul>
</details>

**Review by nadir-ishiguro — APPROVED** [review]
